jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/Ec2MetadataConfigProvider.kt

```diff
@@ -0,0 +1,58 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.credentials.profiles
+
+import software.amazon.awssdk.core.SdkSystemSetting
+import software.amazon.awssdk.profiles.Profile
+import software.amazon.awssdk.profiles.ProfileProperty
+
+object Ec2MetadataConfigProvider {
+    /**
+     * Default IPv4 endpoint for the Amazon EC2 Instance Metadata Service.
+     */
+    private const val EC2_METADATA_SERVICE_URL_IPV4 = "http://169.254.169.254"
+
+    /**
+     * Default IPv6 endpoint for the Amazon EC2 Instance Metadata Service.
+     */
+    private const val EC2_METADATA_SERVICE_URL_IPV6 = "http://[fd00:ec2::254]"
```

Comment:
are these not exposed by the SDK?

Reply:
Nope :/

https://github.com/aws/aws-sdk-java-v2/search?q=http%3A%2F%2F%5Bfd00%3Aec2%3A%3A254%5D

Reply:
let's add comment with link to the java sdk logic then, since most of this looks copied from that